src/gt4py/frontend/gtscript_frontend.py

```diff
@@ -1959,7 +1971,7 @@ def run(self):
             parameters=[
                 parameter_decls[item.name] for item in api_signature if item.name in parameter_decls
             ],
-            computations=computations,
+            computations=[init_computation] + computations if init_computation else computations,
```

Comment:
Maybe you can just return an empty list in `_make_init_computation` to avoid this conditional.